openfe/protocols/openmm_afe/equil_solvation_afe_method.py

```diff
@@ -0,0 +1,575 @@
+            raise NotImplementedError("Can't stitch together results yet")
+
+    def get_individual_estimates(self) -> dict[str, list[tuple[unit.Quantity, unit.Quantity]]]:
+        """
+        Return a dictionary (keyed as `solvent` and `vacuum`) with list
+        of tuples containing the individual free energy estimates and
+        associated MBAR errors for each repeat of the vacuum and solvent
+        calculations.
+
+        Returns
+        -------
+        dGs : dict[str, list[tuple[unit.Quantity, unit.Quantity]]]
+        """
+        vac_dGs = []
+        solv_dGs = []
+
+        for pus in self.data['vacuum'].values():
+            vac_dGs.append((
+                pus[0].outputs['unit_estimate'],
+                pus[0].outputs['unit_estimate_error']
+            ))
+        
+        for pus in self.data['solvent'].values():
+            solv_dGs.append((
+                pus[0].outputs['unit_esitmate'],
+                pus[0].outputs['unit_estimate_error']
+            ))
+
+
+        return {'solvent': solv_dGs, 'vacuum': vac_dGs}
+
+    def get_estimate(self):
+        """Get the solvation free energy estimate for this calculation.
+
+        Returns
+        -------
+        dG : unit.Quantity
+          The solvation free energy. This is a Quantity defined with units.
+        """
+        def _get_average(estimates):
+            # Get the unit value of the first value in the estimates
+            u = estimates[0][0].u
+            # Loop through estimates and get the free energy values
+            # in the unit of the first estimate
+            dGs = [i[0].to(u).m for i in estimates]
+
+            return np.average(dGs) * u
+
+        individual_estimates = self.get_individual_estimates()
+        vac_dG = _get_average(individual_estimates['vacuum'])
+        solv_dG = _get_average(individual_estimates['solvent'])
+
+        return vac_dG - solv_dG
+
+    def get_uncertainty(self):
+        """Get the solvation free energy error for this calculation.
+
+        Returns
+        -------
+        err : unit.Quantity
+          The standard deviation between estimates of the solvation free
+          energy. This is a Quantity defined with units.
+        """
+        def _get_stdev(estimates):
+            # Get the unit value of the first value in the estimates
+            u = estimates[0][0].u
+            # Loop through estimates and get the free energy values
+            # in the unit of the first estimate
+            dGs = [i[0].to(u).m for i in estimates]
+
+            return np.std(dGs) * u
+
+        individual_estimates = self.get_individual_estimates()
+        vac_err = _get_stdev(individual_estimates['vacuum'])
+        solv_err = _get_stdev(individual_estimates['solvent'])
+
+        # return the combined error
+        return np.sqrt(vac_err**2 + solv_err**2)
+
+    def get_forward_and_reverse_analysis(self) -> dict[str, list[dict[str, Union[npt.NDArray, units.Quantity]]]]:
```

Comment:
Not really fond of these nested dictionary returns..

The only atlernative option I could think of here was that we could just have a method for each leg, but then the method signature changes and that might be more jarring / less expected between different protocols? Not sure